### DISCUSSION TITLE
Zigbee Dimmer TS0601 Manufacturer _TZE200_dfxkcots support

### DISCUSSION THREAD

**Discussion by ivanxx** [arendst/Tasmota#11045] [Ideas]
Hi, I've got a couple of these dimmers, supposedly manufactured by Earda and supported by Tasmota (see [here](https://zigbee.blakadder.com/Earda_EDM-1ZAA-EU.html))

The problem is that although I can pair them correctly and a zbstatus2 on the device reports this:

> 12:03:40 CMD: Group 0, Index 2, Command "ZBSTATUS", Data "5"
> 12:03:40 MQT: stat/sonoff_zbbridge/RESULT = {"ZbStatus2":[{"Device":"0x684E","Name":"Isabel_Dimmer","IEEEAddr":"0x60A423FFFE3F5DBE","ModelId":"TS0601","Manufacturer":"_TZE200_dfxkcots","Endpoints":[1],"Config":[]}]}

...pressing the button on the device sends all this data as the bulb goes from 0 to ~50%:

> 12:05:07 ZIG: {"ZbEZSPReceived2":"005F900159004E68BF5D3FFEFF23A460ECD700"}
> 12:05:07 ZIG: ZbEZSPSentRaw 81
> 12:05:07 ZIG: {"ZbEZSPReceived":"59004E68BF5D3FFEFF23A460ECD700"}
> 12:05:07 ZIG: {"ZbEZSPReceived2":"105F9001450000040100EF01010001000062ECD74E68FFFF0A0913010001010100010102"}
> 12:05:07 ZIG: ZbEZSPSentRaw 82
> 12:05:07 ZIG: {"ZbEZSPReceived":"450000040100EF01010001000062ECD74E68FFFF0A0913010001010100010102"}
> 12:05:07 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":61184,"srcaddr":"0x684E","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"LinkQuality":120,"securityuse":0,"seqnumber":98,"fc":"0x09","manuf":"0x0000","transact":19,"cmdid":"0x01","payload":"00010101000101"}}
> 12:05:07 ZIG: ZbZCLRawReceived: {"0x684E":{"EF00/0101":1,"Endpoint":1,"LinkQuality":120}}
> 12:05:07 MQT: tele/sonoff_zbbridge/684E/SENSOR = {"ZbReceived":{"0x684E":{"Device":"0x684E","Name":"Isabel_Dimmer","EF00/0101":1,"Endpoint":1,"LinkQuality":120}}}
> 12:05:08 MQT: tele/sonoff_zbbridge/684E/SENSOR = {"ZbReceived":{"0x684E":{"Device":"0x684E","Name":"Isabel_Dimmer","EF00/0101":1,"Endpoint":1,"LinkQuality":123}}}
> 12:05:08 ZIG: {"ZbEZSPReceived2":"205F900159004E68BF5D3FFEFF23A460ECD700"}
> 12:05:08 ZIG: ZbEZSPSentRaw 83
> 12:05:08 ZIG: {"ZbEZSPReceived":"59004E68BF5D3FFEFF23A460ECD700"}
> 12:05:08 ZIG: {"ZbEZSPReceived2":"305F9001450000040100EF0101000100006BECD74E68FFFF0A0913010001010100010102"}
> 12:05:08 ZIG: ZbEZSPSentRaw 84
> 12:05:08 ZIG: {"ZbEZSPReceived":"450000040100EF0101000100006BECD74E68FFFF0A0913010001010100010102"}
> 12:05:08 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":61184,"srcaddr":"0x684E","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"LinkQuality":120,"securityuse":0,"seqnumber":107,"fc":"0x09","manuf":"0x0000","transact":19,"cmdid":"0x01","payload":"00010101000101"}}
> 12:05:08 ZIG: ZbZCLRawReceived: {"0x684E":{"EF00/0101":1,"Endpoint":1,"LinkQuality":120}}
> 12:05:08 MQT: tele/sonoff_zbbridge/684E/SENSOR = {"ZbReceived":{"0x684E":{"Device":"0x684E","Name":"Isabel_Dimmer","EF00/0101":1,"Endpoint":1,"LinkQuality":120}}}
> 12:05:09 ZIG: {"ZbEZSPReceived2":"405F900159004E68BF5D3FFEFF23A460ECD700"}
> 12:05:09 ZIG: ZbEZSPSentRaw 85
> 12:05:09 ZIG: {"ZbEZSPReceived":"59004E68BF5D3FFEFF23A460ECD700"}
> 12:05:09 ZIG: {"ZbEZSPReceived2":"505F9001450000040100EF0101000100006CECD74E68FFFF0A0913010001010100010102"}
> 12:05:09 ZIG: ZbEZSPSentRaw 86
> 12:05:09 ZIG: {"ZbEZSPReceived":"450000040100EF0101000100006CECD74E68FFFF0A0913010001010100010102"}
> 12:05:09 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":61184,"srcaddr":"0x684E","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"LinkQuality":120,"securityuse":0,"seqnumber":108,"fc":"0x09","manuf":"0x0000","transact":19,"cmdid":"0x01","payload":"00010101000101"}}
> 12:05:09 ZIG: ZbZCLRawReceived: {"0x684E":{"EF00/0101":1,"Endpoint":1,"LinkQuality":120}}
> 12:05:09 MQT: tele/sonoff_zbbridge/684E/SENSOR = {"ZbReceived":{"0x684E":{"Device":"0x684E","Name":"Isabel_Dimmer","EF00/0101":1,"Endpoint":1,"LinkQuality":120}}}
> 12:05:09 ZIG: {"ZbEZSPReceived2":"605F900159004E68BF5D3FFEFF23A460ECD700"}
> 12:05:09 ZIG: ZbEZSPSentRaw 87
> 12:05:09 ZIG: {"ZbEZSPReceived":"59004E68BF5D3FFEFF23A460ECD700"}
> 12:05:09 ZIG: {"ZbEZSPReceived2":"705F9001450000040100EF0101000100006DECD74E68FFFF0A0913010001010100010102"}
> 12:05:09 ZIG: ZbEZSPSentRaw 80
> 12:05:09 ZIG: {"ZbEZSPReceived":"450000040100EF0101000100006DECD74E68FFFF0A0913010001010100010102"}
> 12:05:09 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":61184,"srcaddr":"0x684E","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"LinkQuality":120,"securityuse":0,"seqnumber":109,"fc":"0x09","manuf":"0x0000","transact":19,"cmdid":"0x01","payload":"00010101000101"}}
> 12:05:09 ZIG: ZbZCLRawReceived: {"0x684E":{"EF00/0101":1,"Endpoint":1,"LinkQuality":120}}
> 12:05:09 MQT: tele/sonoff_zbbridge/684E/SENSOR = {"ZbReceived":{"0x684E":{"Device":"0x684E","Name":"Isabel_Dimmer","EF00/0101":1,"Endpoint":1,"LinkQuality":120}}}
> 12:05:09 ZIG: {"ZbEZSPReceived2":"005F900159004E68BF5D3FFEFF23A460ECD700"}
> 12:05:09 ZIG: ZbEZSPSentRaw 81
> 12:05:09 ZIG: {"ZbEZSPReceived":"59004E68BF5D3FFEFF23A460ECD700"}
> 12:05:09 ZIG: {"ZbEZSPReceived2":"105F9001450000040100EF0101000100006EECD74E68FFFF0A0913010001010100010102"}
> 12:05:09 ZIG: ZbEZSPSentRaw 82
> 12:05:09 ZIG: {"ZbEZSPReceived":"450000040100EF0101000100006EECD74E68FFFF0A0913010001010100010102"}
> 12:05:09 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":61184,"srcaddr":"0x684E","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"LinkQuality":120,"securityuse":0,"seqnumber":110,"fc":"0x09","manuf":"0x0000","transact":19,"cmdid":"0x01","payload":"00010101000101"}}
> 12:05:09 ZIG: ZbZCLRawReceived: {"0x684E":{"EF00/0101":1,"Endpoint":1,"LinkQuality":120}}
> 12:05:09 MQT: tele/sonoff_zbbridge/684E/SENSOR = {"ZbReceived":{"0x684E":{"Device":"0x684E","Name":"Isabel_Dimmer","EF00/0101":1,"Endpoint":1,"LinkQuality":120}}}
> 12:05:11 MQT: tele/sonoff_zbbridge/684E/SENSOR = {"ZbReceived":{"0x684E":{"Device":"0x684E","Name":"Isabel_Dimmer","EF00/0101":1,"Endpoint":1,"LinkQuality":123}}}
> 12:05:11 ZIG: {"ZbEZSPReceived2":"205F900159004E68BF5D3FFEFF23A460ECD700"}
> 12:05:11 ZIG: ZbEZSPSentRaw 83
> 12:05:11 ZIG: {"ZbEZSPReceived":"59004E68BF5D3FFEFF23A460ECD700"}
> 12:05:11 ZIG: {"ZbEZSPReceived2":"305F9001450000040100EF01010001000077ECD74E68FFFF0A0913010001010100010102"}
> 12:05:11 ZIG: ZbEZSPSentRaw 84
> 12:05:11 ZIG: {"ZbEZSPReceived":"450000040100EF01010001000077ECD74E68FFFF0A0913010001010100010102"}
> 12:05:11 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":61184,"srcaddr":"0x684E","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"LinkQuality":120,"securityuse":0,"seqnumber":119,"fc":"0x09","manuf":"0x0000","transact":19,"cmdid":"0x01","payload":"00010101000101"}}
> 12:05:11 ZIG: ZbZCLRawReceived: {"0x684E":{"EF00/0101":1,"Endpoint":1,"LinkQuality":120}}
> 12:05:11 MQT: tele/sonoff_zbbridge/684E/SENSOR = {"ZbReceived":{"0x684E":{"Device":"0x684E","Name":"Isabel_Dimmer","EF00/0101":1,"Endpoint":1,"LinkQuality":120}}}
> 12:05:11 ZIG: {"ZbEZSPReceived2":"405F900159004E68BF5D3FFEFF23A460F0D800"}
> 12:05:11 ZIG: ZbEZSPSentRaw 85
> 12:05:11 ZIG: {"ZbEZSPReceived":"59004E68BF5D3FFEFF23A460F0D800"}
> 12:05:11 ZIG: {"ZbEZSPReceived2":"505F9001450000040100EF01010001000078ECD74E68FFFF0A0913010001010100010102"}
> 12:05:11 ZIG: ZbEZSPSentRaw 86
> 12:05:11 ZIG: {"ZbEZSPReceived":"450000040100EF01010001000078ECD74E68FFFF0A0913010001010100010102"}
> 12:05:11 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":61184,"srcaddr":"0x684E","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"LinkQuality":120,"securityuse":0,"seqnumber":120,"fc":"0x09","manuf":"0x0000","transact":19,"cmdid":"0x01","payload":"00010101000101"}}
> 12:05:11 ZIG: ZbZCLRawReceived: {"0x684E":{"EF00/0101":1,"Endpoint":1,"LinkQuality":120}}
> 12:05:11 MQT: tele/sonoff_zbbridge/684E/SENSOR = {"ZbReceived":{"0x684E":{"Device":"0x684E","Name":"Isabel_Dimmer","EF00/0101":1,"Endpoint":1,"LinkQuality":120}}}
> 12:05:11 ZIG: {"ZbEZSPReceived2":"605F900159004E68BF5D3FFEFF23A460F0D800"}
> 12:05:11 ZIG: ZbEZSPSentRaw 87
> 12:05:11 ZIG: {"ZbEZSPReceived":"59004E68BF5D3FFEFF23A460F0D800"}
> 12:05:11 ZIG: {"ZbEZSPReceived2":"705F9001450000040100EF01010001000079ECD74E68FFFF0A0913010001010100010102"}
> 12:05:11 ZIG: ZbEZSPSentRaw 80
> 12:05:11 ZIG: {"ZbEZSPReceived":"450000040100EF01010001000079ECD74E68FFFF0A0913010001010100010102"}
> 12:05:11 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":61184,"srcaddr":"0x684E","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"LinkQuality":120,"securityuse":0,"seqnumber":121,"fc":"0x09","manuf":"0x0000","transact":19,"cmdid":"0x01","payload":"00010101000101"}}
> 12:05:11 ZIG: ZbZCLRawReceived: {"0x684E":{"EF00/0101":1,"Endpoint":1,"LinkQuality":120}}
> 12:05:11 MQT: tele/sonoff_zbbridge/684E/SENSOR = {"ZbReceived":{"0x684E":{"Device":"0x684E","Name":"Isabel_Dimmer","EF00/0101":1,"Endpoint":1,"LinkQuality":120}}}
> 12:05:11 ZIG: {"ZbEZSPReceived2":"005F900159004E68BF5D3FFEFF23A460ECD700"}
> 12:05:11 ZIG: ZbEZSPSentRaw 81
> 12:05:11 ZIG: {"ZbEZSPReceived":"59004E68BF5D3FFEFF23A460ECD700"}
> 12:05:11 ZIG: {"ZbEZSPReceived2":"105F9001450000040100EF0101000100007AECD74E68FFFF0A0913010001010100010102"}
> 12:05:11 ZIG: ZbEZSPSentRaw 82
> 12:05:11 ZIG: {"ZbEZSPReceived":"450000040100EF0101000100007AECD74E68FFFF0A0913010001010100010102"}
> 12:05:11 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":61184,"srcaddr":"0x684E","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"LinkQuality":120,"securityuse":0,"seqnumber":122,"fc":"0x09","manuf":"0x0000","transact":19,"cmdid":"0x01","payload":"00010101000101"}}
> 12:05:11 ZIG: ZbZCLRawReceived: {"0x684E":{"EF00/0101":1,"Endpoint":1,"LinkQuality":120}}
> 12:05:11 MQT: tele/sonoff_zbbridge/684E/SENSOR = {"ZbReceived":{"0x684E":{"Device":"0x684E","Name":"Isabel_Dimmer","EF00/0101":1,"Endpoint":1,"LinkQuality":120}}}
> 

And also, I can send a **_ZbSend {"Device":"0x684E","Send":{"Power":1}}_**, and I get:

> 12:13:43 CMD: ZbSend {"Device":"0x684E","Send":{"Power":1}}
> 12:13:43 SRC: WebConsole from 192.168.200.3
> 12:13:43 CMD: Group 0, Index 1, Command "ZBSEND", Data "{"Device":"0x684E","Send":{"Power":1}}"
> 12:13:43 ZIG: guessing endpoint 1
> 12:13:43 SendZCLCommand_P: zcl_cmd = 
> 12:13:43 ZigbeeZCLSend device: 0x684E, group: 0x0000, endpoint:1, cluster:0x0006, cmd:0x01, send:""
> 12:13:43 ZbSend: shortaddr 0x684E, groupaddr 0x0000, cluster 0x0006, endpoint 0x01, cmd 0x01, data 
> 12:13:43 ZIG: ZbEZSPSend 3400004E6804010600010140010000320103013201
> 12:13:43 ZIG: adding packet to_send, to_ack:5, to_send:5, to_end:5
> 12:13:43 MQT: stat/sonoff_zbbridge/RESULT = {"ZbSend":"Done"}
> 12:13:43 ZIG: Something to_send, to_ack:5, to_send:5, to_end:6
> 12:13:43 ZIG: ZbEZSPSentRaw 506D00013400004E6804010600010140010000320103013201
> 12:13:43 ZIG: {"ZbEZSPReceived2":"066D800134000077"}
> 12:13:43 ZIG: new ack/data received, was 5 now 6
> 12:13:43 ZIG: freeing packet 5 from memory
> 12:13:43 ZIG: ZbEZSPSentRaw 81
> 12:13:43 ZIG: {"ZbEZSPReceived":"34000077"}
> 12:13:43 ZIG: {"ZbEZSPReceived2":"166D900159004E68BF5D3FFEFF23A460ECD700"}
> 12:13:43 ZIG: ZbEZSPSentRaw 82
> 12:13:43 ZIG: {"ZbEZSPReceived":"59004E68BF5D3FFEFF23A460ECD700"}
> 12:13:43 ZIG: {"ZbEZSPReceived2":"266D900159004E68BF5D3FFEFF23A460ECD700"}
> 12:13:43 ZIG: ZbEZSPSentRaw 83
> 12:13:43 ZIG: {"ZbEZSPReceived":"59004E68BF5D3FFEFF23A460ECD700"}
> 12:13:44 ZIG: {"ZbEZSPReceived2":"366D900145000004010600010140010000A2ECD74E68FFFF0508320B01C302"}
> 12:13:44 ZIG: ZbEZSPSentRaw 84
> 12:13:44 ZIG: {"ZbEZSPReceived":"45000004010600010140010000A2ECD74E68FFFF0508320B01C302"}
> 12:13:44 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":6,"srcaddr":"0x684E","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"LinkQuality":120,"securityuse":0,"seqnumber":162,"fc":"0x08","manuf":"0x0000","transact":50,"cmdid":"0x0B","payload":"01C3"}}
> 12:13:44 MQT: tele/sonoff_zbbridge/RESULT = {"ZbResponse":{"Device":"0x684E","Name":"Isabel_Dimmer","Command":"0006!01","Status":195,"StatusMessage":"UNSUPPORTED_CLUSTER","Endpoint":1,"LinkQuality":120}}
> 12:13:44 ZIG: {"ZbEZSPReceived2":"466D90013F00004E680401060001014001000077010000"}
> 12:13:44 ZIG: ZbEZSPSentRaw 85
> 12:13:44 ZIG: {"ZbEZSPReceived":"3F00004E680401060001014001000077010000"}
> 12:13:44 ZIG: ZbEZSPSend 3400004E68040106000101400100003301050033000000
> 12:13:44 ZIG: adding packet to_send, to_ack:6, to_send:6, to_end:6
> 12:13:44 ZIG: Something to_send, to_ack:6, to_send:6, to_end:7
> 12:13:44 ZIG: ZbEZSPSentRaw 656E00013400004E68040106000101400100003301050033000000
> 12:13:44 ZIG: {"ZbEZSPReceived2":"576E800134000078"}
> 12:13:44 ZIG: new ack/data received, was 6 now 7
> 12:13:44 ZIG: freeing packet 6 from memory
> 12:13:44 ZIG: ZbEZSPSentRaw 86
> 12:13:44 ZIG: {"ZbEZSPReceived":"34000078"}
> 12:13:44 ZIG: {"ZbEZSPReceived2":"676E900159004E68BF5D3FFEFF23A460ECD700"}
> 12:13:44 ZIG: ZbEZSPSentRaw 87
> 12:13:44 ZIG: {"ZbEZSPReceived":"59004E68BF5D3FFEFF23A460ECD700"}
> 12:13:44 ZIG: {"ZbEZSPReceived2":"776E900159004E68BF5D3FFEFF23A460ECD700"}
> 12:13:44 ZIG: ZbEZSPSentRaw 80
> 12:13:44 ZIG: {"ZbEZSPReceived":"59004E68BF5D3FFEFF23A460ECD700"}
> 12:13:44 ZIG: {"ZbEZSPReceived2":"076E900145000004010600010100010000A3ECD74E68FFFF0618330100008602"}
> 12:13:44 ZIG: ZbEZSPSentRaw 81
> 12:13:44 ZIG: {"ZbEZSPReceived":"45000004010600010100010000A3ECD74E68FFFF0618330100008602"}
> 12:13:44 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":6,"srcaddr":"0x684E","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"LinkQuality":120,"securityuse":0,"seqnumber":163,"fc":"0x18","manuf":"0x0000","transact":51,"cmdid":"0x01","payload":"000086"}}
> 12:13:44 ZIG: ZbZCLRawReceived: {"0x684E":{"Endpoint":1,"LinkQuality":120}}
> 12:13:44 ZIG: {"ZbEZSPReceived2":"176E90013F00004E680401060001014001000078010000"}
> 12:13:44 ZIG: ZbEZSPSentRaw 82
> 12:13:44 ZIG: {"ZbEZSPReceived":"3F00004E680401060001014001000078010000"}
> 

(I think the above command started being accepted after I made a **_ZbBind {"Device":"0x684E","Cluster":"Illuminance"}_** but note the status message "Unsupported Cluster")

...but sadly it will not light up, and no MQTT command expected to work for a dimmer is accepted either:

> mosquitto_pub -h mqqtserver -u mqttuser  -t "cmnd/sonoff_zbbridge/684E/Power" -m 1
> 12:06:18 MQT: Data Size 3, "cmnd/sonoff_zbbridge/684E/Dimmer 100"
> 12:06:18 SRC: MQTT
> 12:06:18 CMD: Group 0, Index 1, Command "DIMMER", Data "100"
> 12:06:18 MQT: stat/sonoff_zbbridge/RESULT = {"Command":"Unknown"}
> 

Any idea on what can I do? Is this actually an unsupported device? (I'm running on Tasmota 9.0.0.3, BTW)

Thanks for any help!

-Iván.

**Comment by ascillato** [comment]
The version 9.0.0.3 is very old. Please update to latest and try again. Thanks.

**Reply by ivanxx** [reply]
Thanks for the tip, Adrian. I upgraded to latest 9.3.0 and it still does not work, but I've found that somehow tasmota is treating the dimmer like it is a thermostat: 

![screenshot676](https://user-images.githubusercontent.com/3816794/108566299-7bdcd480-7306-11eb-9a9f-e9febdf68f0c.png)

So... How do I tell the Zbbridge that this is actually a dimmer??

Thanks again for the upgrade idea, I thought I was already on a modern version of tasmota!

-Iván.

**Reply by s-hadinger** [reply]
This is unfortunately due to the tuya zigbee non standard protocol. They reuse the same idea for different values and the only way to know what the value is implies knowing what the device is. This is a route where I don't want to go: having a database of each and every device. So will have to find a different way. Let me think about a template like system.

**Reply by ivanxx** [reply]
Great, let me know if I can help in any way. I cloned the repo and tried to find where is it being decided that this device is a thermostat and not a dimmer, but I can see this is not a 5 min. job without previous background in Tasmota.
@ascillato I marked the question as unanswered again, since this device actually pairs but is lacking of proper support. Feel free to mark it as answered again if you think that is more appropriate.

**Reply by s-hadinger** [reply]
Currently it's only recognized as a thermostat. The Dimmer equivalent needs to be added.  You'll need to wait for a couple of weeks unfortunately

**Reply by ivanxx** [reply]
No rush! I'm probably the only owner of these dimmers :-) Thanks for being there!

**Reply by ivanxx** [reply]
HI there, I installed 9.3.1.2 and I see there's no support for these dimmers yet. Anything I can do to help?

Thanks!

**Reply by s-hadinger** [reply]
I will have a look now

**Reply by s-hadinger** [reply]
I tried to simulate what you received and I see:

```
10:10:46.735 RSL: SENSOR = {"ZbReceived":{"0x684E":{"Device":"0x684E","Power":1,"Endpoint":1,"LinkQuality":120}}}
```

So decoding the on/off works. Do you have any more logs, especially when you change the dimmer value?

**Reply by ivanxx** [reply]
I think you nailed it!, see my reply to your next message. If you need the logs though, let me know and I'll post them here.

**Comment by s-hadinger** [comment]
I have added support to send Dimmer to Tuya Zigbee devices. You need though to pair again, or do the following command:
```
ZbRestore {"Device":"0x684E","Name":"Isabel_Dimmer","IEEEAddr":"0x60A423FFFE3F5DBE","ModelId":"TS0601","Manufacturer":"_TZE200_dfxkcots","Endpoints":[1],"Config":["~01.1"]}
```

This marks the device as accepting Tuya commands.

Then you can do `ZbSend {"Device":"0x684E","Write":{"Dimmer":50}}`

Can you please report what range is the Dimmer? Is it 0..254 or 0..100.

**Reply by ivanxx** [reply]
Hi there!  I confirm it is fully working, yay! 

Fun fact: I had some problems after I ZbForgeted the device. I had to retry several times the pairing and hard reset my Sonoff ZbBridge, and finally the Dimmer got registered with a different DeviceID (it is 0x616F now. Was 0x684E)

Tested so far:
ZbSend {"Device":"0x616F","Write":{"Power":0}} (also "Power":1 is working)
ZbSend {"Device":"0x616F","Write":{"Dimmer":0}} the working range is 0-1000 (yes, one thousand, also unexpected but working)

So far it looks like everything is fine, I can't thank you enough!

**Reply by ivanxx** [reply]
The only remaining issue is probably only aesthetic: in the web UI the Dimmer keeps on showing as a thermostat (and target temp reflects the dimmer status, between 0 and 100 (although the range for ZbSend is 0-1000):
![screenshot750](https://user-images.githubusercontent.com/3816794/113683350-63612780-96c4-11eb-91d0-c1fc978ae824.png)

**Reply by ivanxx** [reply]
...and on a side note: when poking around to get the dimmers finally working in OpenHab, I found that sending this specific payload via MQTT:
`mosquitto_pub -h myhost -u myuser -P mypass -t "cmnd/sonoff_zbbridge/ZbSend" -m '{"device":"0x616F", "write":{"Power":""}}'`

...actually **reboots** the bridge! 
I suppose there must be an exception raising and the whole tasmota is terminated. Sending it locally from the bridge console is harmless, but via MQTT it blows up tasmota

**Reply by Marcmk1** [reply]
Wow!! You made my day.. waited almost a year for this one to work!!
Upgraded to the latest version.
First it looked like the new version has an bug. It was Extreem slow.
(It was freezing at the ntp connection) My devices has no direct internet connection.
After fixing the 4 ntp server to my internal (Tasmota console).
I can finally manage this device!!

**Comment by Marcmk1** [comment]
I can Switch the device with Home Assistant. Only the problem is that its light switch itself with 0 and 1 instead of ON - OFF.
Is there a way to change it?
